### PR TITLE
fixing restore domain name bug

### PR DESCRIPTION
**Pull request by calellowitz**
@benrudolph cc: @gcapalbo @snopoke
http://manage.dimagi.com/default.asp?232150#1182380
Im starting to understand the appeal of explicit typing...

### CODE REVIEW

**Comment by czue** [comment]
in the absence of typing there's always tests...

**Comment by benrudolph** [comment]
was going to say, it'd be great to have a test for this. i relied heavily on our testing suite to refactor the restore and it's apparent this case isn't covered

File: corehq/apps/commtrack/util.py [hunk 1]
```diff
@@ -231,11 +231,11 @@ def location_map_case_id(user):
 def get_commtrack_location_id(user, domain):
     if (
         user and
-        user.get_domain_membership(domain.name) and
-        user.get_domain_membership(domain.name).location_id and
+        user.get_domain_membership(domain) and
+        user.get_domain_membership(domain).location_id and
         domain.commtrack_enabled
```
Comment (gcapalbo) [inline corehq/apps/commtrack/util.py]: looks like there are 500s failing on this line (looks like the domain object is necessary)

Comment (gcapalbo) [inline corehq/apps/commtrack/util.py]: nm, just saw https://github.com/dimagi/commcare-hq/pull/12431

**Pull request merged**